src/Frontend/src/components/messages2/StacktraceFormatter.vue

```diff
@@ -0,0 +1,213 @@
+    const match = line.match(atRegex);
+    if (match) {
+      const [, spaces, , methodWithType, paramsWithFile] = match;
+
+      const [type, method] = (() => {
+        const parts = methodWithType.split(".");
+        const method = parts.pop() ?? "";
+        const type = parts.join(".");
+        return [type, method];
+      })();
+
+      const params = paramsWithFile.split(", ").map((param) => {
+        const [paramType, paramName] = param.split(" ");
+        return { name: paramName, type: paramType };
+      });
+
+      const matchFile = line.match(fileAndLineNumberRegEx);
+      let file, lineNumber;
+      if (matchFile) {
+        [, file, lineNumber] = matchFile;
+      }
+
+      return <Node>{ method, type, params, file, lineNumber, spaces };
+    } else {
+      return line;
+    }
+  });
+};
+
+// Process the provided stack trace
+const processStackTrace = (): void => {
+  const rawContent = props.stackTrace;
+  const detectedLanguages = detectLanguagesInOrder(rawContent);
+
+  if (!detectedLanguages.length) {
+    formattedStack.value = [rawContent]; // If no language detected, output raw content
+    return;
+  }
+
+  selectedLanguage.value = detectedLanguages[0]; // Use the first detected language
+  formattedStack.value = formatStackTrace(rawContent, selectedLanguage.value);
+};
+
+watch(
+  () => props.stackTrace,
+  () => {
+    processStackTrace();
+  },
+  { immediate: true }
+);
+</script>
+
+<template>
+  <div class="stack-trace-container">
+    <template v-for="line in formattedStack" :key="line">
+      <template v-if="typeof line === 'string'">
+        <span>{{ line }}</span>
+      </template>
+      <div v-else>
+        {{ line.spaces }}{{ selectedLanguage.at }}
+        <span :class="props.options.frame">
+          <span :class="props.options.type">{{ line.type }}</span
+          >.<span :class="props.options.method">{{ line.method }}</span
+          >(<span :class="props.options.paramsList">
+            <template v-for="(param, index) in line.params" :key="param.name">
+              <span :class="props.options.paramType"> {{ param.type }}</span> <span :class="props.options.paramName">{{ param.name }}</span>
+              <span v-if="index !== line.params.length - 1">, </span>
+            </template> </span
+          >)
+        </span>
+        <template v-if="line.file">
+          {{ selectedLanguage.in }} <span :class="props.options.file">{{ line.file }}</span
+          >:{{ selectedLanguage.line }} <span :class="props.options.line">{{ line.lineNumber }}</span>
+        </template>
+      </div>
+    </template>
+  </div>
+</template>
+
+<style scoped>
+.stack-trace-container {
+  font-family: monospace;
+  white-space: pre-wrap;
+}
+
+.st-frame {
+  color: #007bff;
+}
+
+.st-type {
+  color: #d63384;
+}
+
+.st-method {
+  color: #28a745;
+}
+
+.st-file {
+  color: #fd7e14;
+}
+
+.st-line {
+  color: #6c757d;
+}
+
+.st-param-type {
+  font-style: italic;
+  color: #6f42c1;
+}
```

Comment:
```suggestion
.st-frame {
  color: #00729c;
}

.st-type {
  color: #a11;
}

.st-method {
  color: #164;
}

.st-file {
  color: #c67b3d;
}

.st-line {
  color: #6c757d;
}

.st-param-type {
  font-style: italic;
  color: #6b82ce;
}
```

suggestion to tone down the colours a bit

Reply:
Perfect @PhilBastian